auth_options.go

```diff
@@ -229,25 +228,48 @@ func (opts *AuthOptions) ToTokenV3CreateMap(scope map[string]interface{}) (map[s
 			if opts.ApplicationCredentialSecret == "" {
 				return nil, ErrAppCredMissingSecret{}
 			}
-			// make sure that only one of DomainName or DomainID were provided
-			if opts.DomainID == "" && opts.DomainName == "" {
-				return nil, ErrDomainIDOrDomainName{}
-			}
-			req.Auth.Identity.Methods = []string{"application_credential"}
-			if opts.DomainID != "" {
-				userRequest = userReq{
-					Name:   &opts.Username,
-					Domain: &domainReq{ID: &opts.DomainID},
+			userRequest, err := func(opts *AuthOptions) (*userReq, error) {
```

Comment:
I apologize for being unclear - I meant to get rid of the function entirely. Unless I'm mistaken, the function isn't serving a purpose of being reusable, correct? You could add the contents of the function to be inline with the rest of the series of conditionals and it would be the same? Please let me know if I'm wrong about this.

Reply:
Function is used to simplify the logic. If I remove it, the logic will be full of if then else.

Reply:
Done